src/com/amazon/ion/benchmark/IonMeasurableReadTask.java

```diff
@@ -251,4 +260,26 @@ public void fullyReadDomFromFile(SideEffectConsumer consumer) throws IOException
         ionSystem.newLoader().load(reader);
         reader.close();
     }
+
+    @Override
+    public void fullyReadElementFromBuffer(SideEffectConsumer consumer) throws IOException {
+        sideEffectConsumer = consumer;
+        IonReader reader = readerBuilder.build(buffer);
+        Iterable<AnyElement> elements = elementLoader.loadAllElements(reader);
+        for (AnyElement element : elements) {
+            consumer.consume(element);
+        }
```

Comment:
It occurs to me that we should add a similar loop to `fullyReadDomFromFile` and `fullyReadDomFromBuffer`, both to protect against incorrect optimizations and to provide an apples-to-apples comparison between IonValue and IonElement.